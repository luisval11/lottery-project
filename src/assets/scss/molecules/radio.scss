.radio {
	position: absolute;
	border-radius: 4px;
	background-color: $color-casino-base;
	color: $color-ui-casino-02;
	width: 254px;
	display: flex;
	flex-direction: column;
	margin-top: 32px;
	padding-top: 14px;
	padding-left: 35px;
	padding-bottom: 14px;
	z-index: 101;

	&__wrapper {
		display: flex;
		flex-direction: row;
	}

	&__label {
		cursor: pointer;
		margin-bottom: 15px;
	}
}
.hidden {
	visibility: hidden;
}
