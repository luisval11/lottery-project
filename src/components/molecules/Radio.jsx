import RadioSelected from "../../assets/images/Radio-selected.png";
import RadioUnselected from "../../assets/images/Radio.png";
import { Image } from "../atoms/Image";
import { Input } from "../atoms/Input";
import { Label } from "../atoms/Label";

export const Radio = ({ options, value, onChange }) => {
	return (
		<div className="radio">
			{options.map((option, id) => {
				return (
					<Label
						className={"radio__label"}
						children={
							<>
								<Image
									src={value === option.name ? RadioSelected : RadioUnselected}
								/>
								<Input
									className="hidden"
									key={id}
									type={"radio"}
									checked={value === option.name}
									value={option.name}
									onChange={(event) => {
										const value = event.target.value;
										onChange(value);
									}}
								/>
							</>
						}
						text={option.text}
					></Label>
				);
			})}
		</div>
	);
};
