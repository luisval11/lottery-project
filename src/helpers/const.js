export const URL_LOTTOLAND = "https://www.lottoland.co.uk";
export const SORT_OPTION_NAME = "name";
export const SORT_OPTION_EUR_STAKE_ASC = "stakeEurAsc";
export const SORT_OPTION_EUR_STAKE_DESC = "stakeEurDesc";
export const MAX_GAMES_PER_PAGE = 9;
export const SORT_OPTIONS = [
	{ name: SORT_OPTION_NAME, text: "Name (A-Z)" },
	{ name: SORT_OPTION_EUR_STAKE_ASC, text: "Stake (min to max)" },
	{ name: SORT_OPTION_EUR_STAKE_DESC, text: "Stake (max to min)" },
];
export const EXAMPLE_LISTITEM = {
	LuckyWizard: {
		name: "LuckyWizard",
		displayName: "Lucky Wizard",
		playURL: "/games/luckywizard/play",
		image: "/cms/5bab9d950eb3580fac83392e/Icon_320and250_LuckyWizard.jpg",
		detailURL: "/games/luckywizard",
		currencyData: {
			CHF: {
				noDesk: true,
				noMob: true,
			},
			EUR: {
				minimumStake: 0.1,
			},
			GBP: {
				minimumStake: 0.1,
			},
			SEK: { minimumStake: 1.0 },
			PLN: {},
			JPY: {},
			INR: {},
			NZD: { minimumStake: 0.1 },
			CAD: {},
			HUF: {},
		},
		volatility: ["High"],
		provider: ["RedTiger"],
	},
};
