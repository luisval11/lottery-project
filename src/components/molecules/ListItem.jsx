import imageLogo from "../../assets/images/image.png";
import info from "../../assets/images/info.png";
import { normalizeUrl } from "../../helpers/helpers";
import { Button } from "../atoms/Button";
import { Image } from "../atoms/Image";
import { LinkA } from "../atoms/LinkA";

export const ListItem = ({ id, item, handleSelectItem, selectedItem }) => {
	const { displayName, provider, currencyData } = item;
	const image = normalizeUrl(item.image);
	const playURL = normalizeUrl(item.playURL);

	const handleOnClick = () => {
		handleSelectItem(id);
	};

	return (
		<div
			className={`list-item__wrapper-rectangle${
				selectedItem === id ? "--selected" : ""
			}`}
		>
			<div
				className={`list-item${selectedItem === id ? "--selected" : ""}`}
				onClick={handleOnClick}
			>
				<div className="list-item__wrapper">
					<div>
						<Image
							className={`list-item__info-image${
								!(selectedItem === id) ? "--hidden" : ""
							}`}
							src={info}
							alt="Information"
						/>
						<Image
							className="list-item__image"
							src={image}
							alt={displayName}
						/>
					</div>
					<div className="list-item__info">
						{provider && <label>{provider.join(",")}</label>}
						{displayName && <h5>{displayName}</h5>}
						{currencyData &&
							currencyData.EUR &&
							currencyData.EUR.minimumStake && (
								<label>â‚¬{currencyData.EUR.minimumStake} min. Stake</label>
							)}
					</div>
				</div>
				{playURL && (
					<LinkA
						className={`list-item__button${
							!(selectedItem === id) ? "--hidden" : ""
						}`}
						href={playURL}
						children={<Button className="list-item__link-button" text="Play" />}
					/>
				)}
			</div>
			<div
				className={`list-item__item-separator${
					selectedItem === id ? "--selected" : ""
				}`}
			/>
		</div>
	);
};
