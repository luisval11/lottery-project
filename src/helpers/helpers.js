import { SORT_OPTION_EUR_STAKE_ASC, SORT_OPTION_EUR_STAKE_DESC, SORT_OPTION_NAME, URL_LOTTOLAND } from "./const";

export const normalizeUrl = (endpoint) => {
	return URL_LOTTOLAND + endpoint;
};

export const isValidGame = (game) => {
	return (
		game.currencyData != undefined &&
		game.currencyData.EUR != undefined &&
		game.currencyData.EUR.minimumStake != undefined
	);
};

export const getItemsToShow = (sort, initialValues, initial, size) => {
	switch (sort) {
		case SORT_OPTION_NAME:
			return Object.values(initialValues)
				.filter((game) => isValidGame(game))
				.sort((gameA, gameB) => {
					return gameA.displayName.localeCompare(gameB.displayName);
				})
				.slice(initial, size);

		case SORT_OPTION_EUR_STAKE_ASC:
			return Object.values(initialValues)
				.filter((game) => isValidGame(game))
				.sort((gameA, gameB) => {
					return (
						gameA.currencyData.EUR.minimumStake -
						gameB.currencyData.EUR.minimumStake
					);
				})
				.slice(initial, size);

		case SORT_OPTION_EUR_STAKE_DESC:
			return Object.values(initialValues)
				.filter((game) => isValidGame(game))
				.sort((gameA, gameB) => {
					return (
						gameB.currencyData.EUR.minimumStake -
						gameA.currencyData.EUR.minimumStake
					);
				})
				.slice(initial, size);
	}
};
