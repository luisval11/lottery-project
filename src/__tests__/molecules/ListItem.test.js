import { h } from "preact";
import { mount, shallow } from "enzyme";
import { ListItem } from "../../components/molecules/ListItem";
import { EXAMPLE_LISTITEM } from "../../helpers/const";

describe("Test on ListItem", () => {
	let id = 0;
	const item = EXAMPLE_LISTITEM.LuckyWizard;
	let selectedItem = null;
	const handleSelectItem = (value) => {
		selectedItem = value;
	};

	test("should match initial snapshot", () => {
		let wrapper = shallow(
			<ListItem
				key={id}
				id={id}
				item={item}
				handleSelectItem={handleSelectItem}
				selectedItem={selectedItem}
			/>
		);
		expect(wrapper).toMatchSnapshot();
	});

	test("should have 2 img, according to designs to Figma (image of item and info icon)", () => {
		const countImage = 2;
		let wrapper = mount(
			<ListItem
				key={id}
				id={id}
				item={item}
				handleSelectItem={handleSelectItem}
				selectedItem={selectedItem}
			/>
		);
		expect(wrapper.find("img").length).toBe(countImage);
	});

	test("should have 1 LinkA, according to designs to Figma (wrapper of button to go the external item)", () => {
		const countLinkA = 1;
		let wrapper = mount(
			<ListItem
				key={id}
				id={id}
				item={item}
				handleSelectItem={handleSelectItem}
				selectedItem={selectedItem}
			/>
		);
		expect(wrapper.find("a").length).toBe(countLinkA);
	});

	test("should have 2 Label, according to designs to Figma (provider of the item and minimum stake in EUR currency)", () => {
		const countLabel = 2;
		let wrapper = mount(
			<ListItem
				key={id}
				id={id}
				item={item}
				handleSelectItem={handleSelectItem}
				selectedItem={selectedItem}
			/>
		);
		expect(wrapper.find("label").length).toBe(countLabel);
	});

	test("should have 1 Button, according to designs to Figma (button to go the external item)", () => {
		const countButton = 1;
		let wrapper = mount(
			<ListItem
				key={id}
				id={id}
				item={item}
				handleSelectItem={handleSelectItem}
				selectedItem={selectedItem}
			/>
		);
		expect(wrapper.find("button").length).toBe(countButton);
	});

	test("should have 1 h5, according to designs to Figma (heading for displayName of item)", () => {
		const countHeading = 1;
		let wrapper = mount(
			<ListItem
				key={id}
				id={id}
				item={item}
				handleSelectItem={handleSelectItem}
				selectedItem={selectedItem}
			/>
		);
		expect(wrapper.find("h5").length).toBe(countHeading);
	});

	test("className should change to 'selected' when clicks on the listItem", () => {
		let wrapper = mount(
			<ListItem
				key={id}
				id={id}
				item={item}
				handleSelectItem={handleSelectItem}
				selectedItem={selectedItem}
			/>
		);

		//Extract the onClick for the div input
		const { onClick } = wrapper.find("div.list-item").props();
		onClick(id);

		//Forces a rerender of the component with the new value after onClick;
		wrapper.setProps({
			key: id,
			id: id,
			item: item,
			handleSelectItem: handleSelectItem,
			selectedItem: id,
		});
		wrapper.update();

		expect(wrapper.find("div.list-item--selected").length).toBe(1);
		expect(wrapper.find("div.list-item").length).toBe(0);
	});
});
