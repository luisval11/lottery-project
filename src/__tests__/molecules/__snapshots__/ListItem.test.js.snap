// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on ListItem should match initial snapshot 1`] = `ShallowWrapper {}`;
